### ISSUE TITLE
Compare a small set of loss and objective choices for Deep Sets

### ISSUE THREAD

**Issue by annawoodard** [dsi-clinic/vanguard#123]
**Why this issue exists**

The current Deep Sets trainer always uses one weighted binary cross-entropy setup. That is a fine baseline, but we should check whether a small number of alternative objectives behave better on this class-imbalanced problem.

**Main goal**

Add a small, manageable set of objective choices and compare them fairly.

**Recommended starting set**

Start with:

- current weighted `BCEWithLogitsLoss`
- unweighted `BCEWithLogitsLoss` as a reference
- focal loss as the main alternative

Optional stretch:

- label smoothing
- threshold tuning on training folds after probability prediction

**Student-friendly advice**

Keep this question simple:

- does a different training objective make validation performance better or more stable?

If a new loss only changes the number a tiny amount but makes training much harder to explain, that may not be worth it.

**Implementation guidance**

- add config keys for objective choice
- keep the optimizer and architecture fixed at first
- compare losses on the same point-feature dataset and same folds

**Definition of done**

- loss choice is configurable
- at least one alternative loss beyond the baseline is implemented
- results are summarized in one small table
- one default loss choice is selected

**Helpful file pointers**

- `train_deepsets.py`
- `deepsets_model.py`
- `configs/deepsets_ispy2.yaml`

**Comment by Nedak23** [comment]
## Loss Function Comparison Results

Ran all 3 loss variants on the ISPY2 cohort

| Loss | Mean AUC | Std AUC |
|---|---|---|
| weighted_bce | 0.532 | 0.029 |
| unweighted_bce | 0.541 | 0.026 |
| **focal** | **0.546** | **0.020** |

<img width="1600" height="800" alt="Image" src="https://github.com/user-attachments/assets/fd556f26-bdd9-4187-8aba-7c733c46cab7" />

Focal loss has the highest mean AUC and the lowest variance. The differences are modest (~1.4 AUC points between focal and weighted BCE), but focal is consistently more stable across folds.

**Selected default: `focal`** — marginally better mean AUC and noticeably tighter fold variance. Updated `config.py` default
